npc/rule.go

```diff
@@ -32,13 +31,13 @@ func newRuleSpec(policyType policyType, proto *string, srcHost, dstHost ruleHost
 		args = append(args, "-p", *proto)
 	}
 	srcComment := "anywhere"
-	if srcHost != nil && !srcHost.isNil() {
+	if srcHost != nil && !reflect.ValueOf(srcHost).IsNil() {
```

Comment:
what's going on here?  how does `srcHost` git to be nil-valued but not `nil` ?

Reply:
Go :man_shrugging: https://play.golang.org/p/bCC2ztY1NBh

Reply:
ok, it is `nil`, but of a different type to the one you asked about.